lib/charms/filesystem_client/v0/filesystem_info.py

```diff
@@ -217,12 +217,17 @@ def __init__(
     ) -> None:
         if not scheme:
             raise FilesystemInfoError("scheme cannot be empty")
-        if len(hosts) == 0:
+        str_hosts = []
+        for host in hosts:
+            if not host:
+                raise FilesystemInfoError(f"invalid host `{host}`")
+            str_hosts.append(str(host))
+        if len(str_hosts) == 0:
```

Comment:
An empty list `[]` is considered a falsey value in Python, so if there are no provided hosts, you could just check for the falsey-ness of `str_hosts`:

```suggestion
        if not str_hosts:
```